staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -157,7 +158,7 @@ func (r *withRetry) IsNextRetry(ctx context.Context, restReq *Request, httpReq *
 	r.retryAfter.Wait = time.Duration(seconds) * time.Second
 	r.retryAfter.Reason = getRetryReason(r.attempts, seconds, resp, err)
 
-	if err := r.prepareForNextRetry(ctx, restReq); err != nil {
+	if err := r.prepareForNextRetry(ctx, restReq, closer); err != nil {
```

Comment:
don't we have to close it always that we call `r.prepareForNextRetry`?

Reply:
maybe we can unconditionally rewind here

Reply:
there are several paths in prepareForNextRetry that don't

Reply:
are you sure? I see that at this point we don't have to do anything anymore with the response

```
	// if we are here, we have either a or b:
	//  a: we have a retryable error, for which we already
	//     have an artificial "Retry-After" response.
	//  b: we have a response from the server for which we
	//     need to check if it is retryable
	seconds, wait := checkWait(resp)
	if !wait {
		return false
	}
```

we are after the wait